src/RA_Dlg_Memory.cpp

```diff
@@ -29,18 +29,18 @@ HWND Dlg_Memory::m_hWnd = nullptr;
 HFONT MemoryViewerControl::m_hViewerFont = nullptr;
 SIZE MemoryViewerControl::m_szFontSize;
 unsigned int MemoryViewerControl::m_nDataStartXOffset = 0;
-unsigned int MemoryViewerControl::m_nAddressOffset = 0;
+ra::ByteAddress MemoryViewerControl::m_nAddressOffset = 0;
 unsigned int MemoryViewerControl::m_nWatchedAddress = 0;
 MemSize MemoryViewerControl::m_nDataSize = MemSize::EightBit;
-unsigned int MemoryViewerControl::m_nEditAddress = 0;
+ra::ByteAddress MemoryViewerControl::m_nEditAddress = 0;
 unsigned int MemoryViewerControl::m_nEditNibble = 0;
 bool MemoryViewerControl::m_bHasCaret = 0;
 unsigned int MemoryViewerControl::m_nCaretWidth = 0;
 unsigned int MemoryViewerControl::m_nCaretHeight = 0;
 unsigned int MemoryViewerControl::m_nDisplayedLines = 8;
 unsigned short MemoryViewerControl::m_nActiveMemBank = 0;
 
-unsigned int m_nPage = 0;
+std::size_t m_nPage = 0;
```

Comment:
can you explain why you use `gsl::index` in some places and `size_t` in others?

Reply:
In 64-bit based architectures (usually still varies a bit by micro), a user-mode application can't index past `PTRDIFF_MAX`. The result of pointer math is also `ptrdiff_t`. `gsl::index` is an alias of `ptrdiff_t `(which is either `int` or `long long` depending on the platform). I wasn't sure if this was supposed to be an index or not. It was already unsigned, because of truncation it needed to be different but still preserve the type for x86/Win32.